integration/src/main/java/ai/djl/integration/tests/ndarray/NDIndexTest.java

```diff
@@ -52,6 +54,23 @@ public void testPick() {
         }
     }
 
+    @Test
+    public void testGather() {
+        // Currently in windows gradle cannot find all the engines.
+        // In the dependency, changing runtimeOnly to api however will remedy the problem.
+        // TODO: remove this when gradle problem is fixed.
+        TestRequirements.notWindows();
+        Engine engine = Engine.getEngine("PyTorch");
```

Comment:
The integration module is supposed to be for "engine agnostic tests". So, you shouldn't need to specify "PyTorch" here. For the other engines, it will throw a `UnsupportedOperationException` instead of running successfully. But, our test runner is designed to treat `UnsupportedOperationException` as a special test result, not a failure. It's similar to how a testng `SkipException` counts as a skipped test result rather than a test failure.

In the future, we should be able to implement gather for the other engines and then it will just run the test because it is already there.